### ISSUE TITLE
[Question] Multiple prediction result columns for text classification

### ISSUE THREAD

**Issue by shaggygi** [dotnet/machinelearning-modelbuilder#1510]
I came across somewhere that the API now supports multiple prediction result columns for the text classification.  Is there any news/updates when this might be added to the Visual Studio UI?  If not, is there any docs that explain how to use with API for now?  Any info is appreciated.  Thx

**Comment by LittleLittleCloud** [comment]
Hi @shaggygi 

Thanks for the ask, I'll try to help you here. Can you add more details about `multiple prediction result columns` though? Do you mean multi-lable classification?

**Comment by shaggygi** [comment]
Thanks @LittleLittleCloud  and be aware I'm still very new to ML.NET.

I was successful in setting up a console app, a mbconfig for Text Classification model, reading from a CSV file, and evaluating results.  In the Data tab, there is a "Column to predict" label.  If I understand, there is a future VS update on being able to select multiple labels to predict.

In my scenario related to cause of failure, it would be very helpful to breakdown and show percentages based on the multiple columns predicted.  For example, I have Failure Reason, Item Number, and Item Location columns.  With the current one label to select/predict, I can only see for Failure Reason.  Instead, it would be nice like the following...

(disclaimer info is only for example)
| Failure Reason | Item Number | Item Location | Predict % |
| --------------- | --------------- | --------------- | --------------- |
| Analog Input | AAAA | 102, 200 | 54% |
| Analog Input | BBBB | 422, 338 | 32% |
| Not Found |  |  | 26% |
| Communication | CCDD | 1, 335 | 2% |

Hopefully this helps.  Thanks again.

**Comment by LittleLittleCloud** [comment]
Just to make sure I understand your ask, your dataset has three columns: `Failure Reason`, `Item Number`, `Item Location`. The `Failure Reason` is the label for classification, `Item Number` and `Item Location` are the feature columns. Is that correct? Or your dataset has more than three columns and you want to select those three columns as predict columns?

**Comment by shaggygi** [comment]
Yes, I have more columns in the dataset.  Sorry about leaving that info out.  I have about 5 other fields used as parameters and currently only predict `Failure Reason`, but looking to add `Item Number` and `Item Location` to the predicted columns.  Thx

**Comment by LittleLittleCloud** [comment]
So the [multi-label classification](https://en.wikipedia.org/wiki/Multi-label_classification) should be the one that you look into.

@briacht @JakeRadMSFT  We have one more ask for multi-label classification, do we want to add this to our roadmap?

**Comment by LittleLittleCloud** [comment]
#1414

**Comment by briacht** [comment]
Yes! I definitely think we should add this to our roadmap.

**Comment by shaggygi** [comment]
This would be super helpful if you added.  I'll upvote on the provided link.  Feel free to close this since it appears it might be a duplicate.  I'll subscribe and try watching out for a possible update.  Thanks again for considering this 👍

**Comment by Arvind2121** [comment]
Hey i was looking into this issue what is the progress now is there any way to have multiple prediction column in ML.NET ?
